app.rb

```diff
@@ -214,14 +214,17 @@ def validate!
       @created = true
       @script = attr[:script]
       job = Job.new(
+        arguments: attr[:arguments],
+        array: attr[:array],
```

Comment:
Currently the `array` input is missing its documentation which has hidden an idiosyncrasy. The `array` input is a string in csv format where the `arguments` is a literal array of strings.

We need to document this difference to make it easier on the client to craft the request.

It might be worth considering submitting the "arrays" in a standardised format (e.g. either csv or literal json array).

Reply:
The format that slurm supports for the `--array` parameter is not as simple as I've implemented here.  In order to support migration from slurm, we'd like to be backwards compatible on as many options as possible, particularly the more common options.

As such, I expect that `--array` will eventually support much of the slurm syntax.  Some examples are:

* `--array 0-3` == `--array 0,1,2,3`
* `--array 0,1,4-5` == `--array 0,1,4,5`
* `--array 0-15:4` == `--array 0,4,8,12`

This shorthand is obviously not suitable for the `arguments` parameter.

Whilst it might seem tempting to expand the shorthand client-side and then submit the arrays in a common format, that is not suitable for large arrays.  I would much sooner send the string `0-1023` than I would its expanded version.  Likewise, I'd much soon read the string `0-8,10-15,18-23` in the logs than its expanded version.

Reply:
Okay thanks for the clarification. Possible the API attribute `array` not being an array is the source of the confusion. The CLI `--array` input doesn't necessarily need parody as the client can handle the key translation.

Possible `array_range` (or something to that effect) would be clearer. However this is me being a :panda_face: